node-lib/src/runner.rs

```diff
@@ -253,8 +271,40 @@ async fn start(
     .expect("Failed to prepare data directory");
     let lock_file = lock_data_dir(&data_dir)?;
 
+    if run_options.clean_data.unwrap_or(false) {
+        clean_data_dir(&data_dir, &data_dir.join(LOCK_FILE_NAME))?;
+    }
+
     log::info!("Starting with the following config:\n {node_config:#?}");
-    let (manager, controller) = initialize(chain_config, data_dir, node_config).await?;
+    let (manager, controller) = match initialize(
+        chain_config.clone(),
+        data_dir.clone(),
+        node_config.clone(),
+    )
+    .await
+    {
+        Ok((manager, controller)) => (manager, controller),
+        Err(error) => match error.downcast_ref::<ChainstateError>() {
+            Some(ChainstateError::FailedToInitializeChainstate(e)) => match e {
+                chainstate::InitializationError::ChainstateStorageVersionMismatch(_, _) => {
+                    log::info!("Failed to init chainstate: {e} \n Cleaning up current db and trying from scratch.");
+
+                    let storage_config: StorageBackendConfig =
+                        node_config.chainstate.clone().unwrap_or_default().storage_backend.into();
+
+                    // cleanup storage directory and retry initialization
+                    if let Some(storage_subdir_name) = storage_config.subdirectory_name() {
+                        let path = data_dir.join(storage_subdir_name);
+                        std::fs::remove_dir_all(path).expect("Removing directory must succeed");
```

Comment:
This can fail if the directory doesn't exist. 

Also, please write an error message that's more friendly. This message directly faces users at a very high level if something fails.